core/src/com/unciv/logic/civilization/Civilization.kt

```diff
@@ -865,6 +865,22 @@ class Civilization : IsPartOfGameInfoSerialization {
             // move new capital
             city.cityConstructions.addBuilding(city.capitalCityIndicator())
             city.isBeingRazed = false // stop razing the new capital if it was being razed
+
+            // move the buildings with MovedToNewCapital unique
+            if (oldCapital != null) {
+                // Get the Sequence of the buildings to move
+                val buildingsToMove = oldCapital.cityConstructions.getBuiltBuildings().filter {
+                    it.hasUnique(UniqueType.MovedToNewCapital)
+                }
```

Comment:
The .toSet() needs to be *here* - otherwise, once we come to "remove the buildings", we evaluate this a second time and see that hey, there are no such buildings in the old capital, look at that!

Reply:
Oh I see SomeTrog already mentioned this